lib/OpenAI/API/Request.pm

```diff
@@ -160,19 +178,13 @@ sub _request_headers {
 sub _send_request {
     my ( $self, $req ) = @_;
 
-    my $cond_var = AnyEvent->condvar;
+    my $loop = IO::Async::Loop->new();
 
-    $self->_async_http_send_request($req)->then(
-        sub {
-            $cond_var->send(@_);
-        }
-    )->catch(
-        sub {
-            $cond_var->send(@_);
-        }
-    );
+    my $future = $self->_async_http_send_request($req);
+
+    $loop->await($future);
```

Comment:
This one line here is forcing the future to block until it's ready; so basically you're defeating the entire point of trying to act asynchronously anyway.

Reply:
That's correct -- there is a `_send_request` method, for those who want a *synchronous* request; and `_send_request_async`, for those who want an *asynchronous* request.